packages/components/src/slot-fill/context.ts

```diff
@@ -1,20 +1,22 @@
 /**
  * WordPress dependencies
  */
+import { observableMap } from '@wordpress/compose';
 import { createContext } from '@wordpress/element';
+
 /**
  * Internal dependencies
  */
 import type { BaseSlotFillContext } from './types';
 
 const initialValue: BaseSlotFillContext = {
+	slots: observableMap(),
+	fills: observableMap(),
 	registerSlot: () => {},
 	unregisterSlot: () => {},
 	registerFill: () => {},
 	unregisterFill: () => {},
-	getSlot: () => undefined,
-	getFills: () => [],
-	subscribe: () => () => {},
```

Comment:
`getSlot` and `getFills` are replaced by simple map lookups: `slots.get( name )`. The `subscribe` method is also a method of the observable map now.